File: frontend/src/api/knowledge.ts
import request, { type ApiResponse } from './request'

export type ReviewExpiryStatus = 'normal' | 'upcoming' | 'overdue' | null

export interface KnowledgeItem {
  id: number
  title: string
  content: string
  category_id: number
  category_name?: string
  submitter_id: number
  submitter_name?: string
  review_status: 'pending' | 'approved' | 'rejected'
  reject_reason?: string
  is_read?: boolean
  suggested_review_cycle?: string
  next_review_date?: string
  review_expiry_status?: ReviewExpiryStatus
  created_at: string
  updated_at: string
}

export interface KnowledgeListResponse {
  items: KnowledgeItem[]
  total: number
  page: number
  page_size: number
}

export interface KnowledgeCreate {
  title: string
  content: string
  category_id: number
  suggested_review_cycle?: string
}

export interface KnowledgeUpdate {
  title?: string
  content?: string
  category_id?: number
  suggested_review_cycle?: string
}

export const getKnowledgeList = (params: {
  category_id?: number
  page?: number
  page_size?: number
  review_status?: string
  keyword?: string
  review_expiry_status?: string
}) => {
  return request.get<any, ApiResponse<KnowledgeListResponse>>('/knowledge', { params })
}

export const getMyKnowledge = (params: {
  page?: number
  page_size?: number
  review_expiry_status?: string
}) => {
  return request.get<any, ApiResponse<KnowledgeListResponse>>('/knowledge/my', { params })
}

export const getKnowledgeDetail = (id: number) => {
  return request.get<any, ApiResponse<KnowledgeItem>>(`/knowledge/${id}`)
}

export const createKnowledge = (data: KnowledgeCreate) => {
  return request.post<any, ApiResponse<KnowledgeItem>>('/knowledge', data)
}

export const updateKnowledge = (id: number, data: KnowledgeUpdate) => {
  return request.put<any, ApiResponse<KnowledgeItem>>(`/knowledge/${id}`, data)
}

export const deleteKnowledge = (id: number) => {
  return request.delete<any, ApiResponse>(`/knowledge/${id}`)
}

File: backend/app/schemas/knowledge.py
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime


class KnowledgeBase(BaseModel):
    title: str = Field(..., description="标题")
    content: str = Field(..., description="内容")
    category_id: int = Field(..., description="分类ID")


class KnowledgeCreate(KnowledgeBase):
    pass


class KnowledgeUpdate(BaseModel):
    title: Optional[str] = Field(None, description="标题")
    content: Optional[str] = Field(None, description="内容")
    category_id: Optional[int] = Field(None, description="分类ID")


class KnowledgeResponse(BaseModel):
    id: int
    title: str
    content: str
    category_id: int
    category_name: Optional[str] = None
    submitter_id: int
    submitter_name: Optional[str] = None
    review_status: str
    reject_reason: Optional[str]
    is_read: Optional[bool] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class KnowledgeListResponse(BaseModel):
    items: List[KnowledgeResponse]
    total: int
    page: int
    page_size: int
